### ISSUE TITLE
Home Unclear Action Bar Control Purpose 

### ISSUE THREAD

**Issue by johnhbenetech** [benetech/MathShare#750]
The controls in the action bar are evidently intended to assist students with sharing a problem set, or somehow linking it into some external services.  Unfortunately, the accessible labels and text present in the bar do not effectively indicate the purpose of this area of the page, and a screen reader user is likely to be unclear about what they should or can use it for. 
 
First, the action bar text does not effectively indicate what the controls are for, nor is it implemented as a heading.  It reads “Assign:” when a problem set is being edited or created, and “Submit:” when a problem set is being viewed.  This is likely to cause a user to ask questions such as, “Submit to where?”, “Assign to whom?”, etc.  It’s also not clear why the text differs based on the action in the URL — why would a problem set be assigned to an external classroom while it was being created, but submitted to it while being viewed? 
 
Next, the button to view the problem set permalink in a modal only has an accessible label of “Link”.  This single-word text could be construed both as a verb (to link an account) or as a noun (a permalink).  The latter is intended here, but cannot be gauged from the label text alone. 
 
Finally, the links to submit or assign a problem set to Google Classroom or Microsoft Teams exhibit a number of accessibility concerns:

* The controls are links, implemented as `<button>` elements with their roles overridden.
* A screen reader will read the label twice, because the buttons have child text content as well as a `<img>` with a non-empty alt attribute.
* The label text is insufficient to describe the destination or purpose of the links.  It contains the name of the external service (Google Classroom or Microsoft Teams), but no information about what action the user will trigger on the particular platform when they activate the link.
* The text content inside the button is contained inside a `<div>`, which is not a valid child element of `<button>`.
* The links cause a new browser window or tab to be opened when activated, but this is not accessibly indicated to any users (see #744).

#### Suggested Solution

As suggested in #749, it may be appropriate to move the controls out of the action bar and into the drop-down menu after the page heading.  If that advice is not followed, the current action bar text should be marked up as a level 2 heading (using a native `<h2>` element).  The text itself should also be modified to more clearly indicate the purpose of the controls in the bar.

Regardless of where the action bar content is positioned, the labels of the controls must be modified so that they accurately and unambiguously describe their purpose.

* The button with the current label of “Link” should have a label of “Show Permalink”, “Share Permalink” or similar.
* The Microsoft Teams and Google Classroom links should have labels which clearly describe their associated action on the target platform.  Specifically:
    * If the action differs based on the current context, such as viewing or editing a problem set, the link text should reflect this difference.
    * The link text should be understandable without any additional context being externally provided (such as a teacher describing how to use the links).
    * The images inside the links should be marked as decorative, if text content is present to provide an accessible label already.  This can be accomplished by supplying an empty string (without whitespace) as the value of the `alt` attribute.

Aha! Link: https://diagramlabs.aha.io/features/B-132

**Comment by johnhbenetech** [comment]
Suggested Solution 
 
As suggested in the previous section of this document (“Action Bar Content Positioned Before Main Heading”), it may be appropriate to move the controls out of the action bar and into the drop-down menu after the page heading.  If that advice is not followed, the current action bar text should be marked up as a level 2 heading (using a native < h2 > element).  The text itself should also be modified to more clearly indicate the purpose of the controls in the bar. 
 
Regardless of where the action bar content is positioned, the labels of the controls must be modified so that they accurately and unambiguously describe their purpose. 
 
* The button with the current label of “Link” should have a label of “Show Permalink”, “Share Permalink” or similar. 
* The Microsoft Teams and Google Classroom links should have labels which clearly describe their associated action on the target platform.  Specifically: 

o If the action differs based on the current context, such as viewing or editing a problem set, the link text should reflect this difference. 

o The link text should be understandable without any additional context being externally provided (such as a teacher describing how to use the links). 

o The images inside the links should be marked as decorative, if text content is present to provide an accessible label already.  This can be accomplished by supplying an empty string (without whitespace) as the value of the alt attribute.

**Comment by jscholes** [comment]
@rupeshparab - this has been significantly cleaned up, which is great.

For some reason, there are `<h2>` elements nested within the three action bar buttons.  This is not valid HTML, because a `<button>` can only wrap non-interactive phrasing content.  Headings don't fall into that category.

They won't be rendered for screen reader users, but I do want to stress that the buttons don't need headings regardless.  These should be removed.

**Comment by rupeshparab** [comment]
I have replaced h2 inside buttons with span

**Comment by jscholes** [comment]
Fixed in https://github.com/benetech/MathShare/commit/e2ce33339dacf29c4f69d44151d83fc56b53ccaa.
